src/components/Avatar/Avatar.style.ts

```diff
@@ -0,0 +1,84 @@
+import { styled } from 'styled-components';
+
+export type Shape = 'circle' | 'round' | 'square';
+export type Size = 'XS' | 'S' | 'M' | 'L' | 'XL';
+
+export interface AvatarProps {
+  size?: Size;
+  shape?: Shape;
+  src?: string;
+  isBorder?: boolean;
+  isShadow?: boolean;
+  isEdit?: boolean;
+  onClick?: (event: React.MouseEventHandler<HTMLElement>) => void;
+}
+
+const ShapeToStyle: { [key: string]: string } = {
+  circle: '50%',
+  round: '5px',
+  square: '0px',
+};
+
+const SizeToStyle: { [key: string]: string } = {
+  XS: '26px',
+  S: '36px',
+  M: '46px',
+  L: '56px',
+  XL: '66px',
+};
+
+export const AvatarWrapper = styled.div<{
+  size?: Size;
+  shape?: Shape;
+  src?: string;
+  $isBorder?: boolean;
+  $isShadow?: boolean;
+}>`
+  position: relative;
+  box-sizing: border-box;
+  display: inline-block;
+  width: ${({ size }) => size && SizeToStyle[size]};
+  height: ${({ size }) => size && SizeToStyle[size]};
+  overflow: hidden;
+  border: ${({ $isBorder, theme }) =>
+    $isBorder ? `1px solid ${theme.transparent_30}` : 'none'};
+  border-radius: ${({ shape }) => shape && ShapeToStyle[shape]};
+  box-shadow: ${({ $isShadow, theme }) =>
+    $isShadow &&
+    (theme.mode === 'light'
+      ? `0px 4px 8px ${theme.black_primary}20`
+      : `0px 4px 8px #00000050`)};
+  &::before {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    content: '';
+    background-color: ${({ theme }) => theme.background_primary};
+    background-image: ${({ src, theme }) =>
+      src
+        ? `url(${src})`
```

Comment:
p5;
`src` 경로가 주어졌지만 경로로부터 image를 불러오지 못한 예외 처리에 대한 렌더링도 필요할 거 같습니다!
아래와 같이 url 경로에 추가 예외 url이 있으면 대체되는 거 같습니다.
[관련 레퍼런스](https://velog.io/@whddnjs1715/img-tag-onError)도 참고해보시면 좋을 거 같아요!
```suggestion
        ? `url(${src}), ${
            theme.mode === 'light'
              ? 'url(src/assets/avatar-light.png)'
              : 'url(src/assets/avatar-dark.png)'
          }`
```  
<br>

그리고 중복되는 부분은 `altImage`라는 변수로 따로 빼도 될 거 같습니다.
```ts
const altImage = theme.mode === 'light'
              ? 'url(src/assets/avatar-light.png)'
              : 'url(src/assets/avatar-dark.png)';
```
```suggestion
        ? `url(${src}), ${altIamge}`
```

Reply:
오 진짜 유용한 정보에요 감사합니다! 레퍼런스에는 img 태그를 권장하지만 background-image로도 이렇게 처리가 되는군요! 감사합니다~~

Reply:
altImage의 경우 Avatar 컴포넌트에서 적용해보려고 했는데 styled-component에서 대체 url로 예외처리를 해줄 경우, 중복으로 예외처리가 됩니다. 따라서 아래 수현님 아이디어로 적용했습니다! 😄 좋은 의견 감사합니다~ 레퍼런스👍👍👍

Reply:
예외 처리 코드는 다음과 같습니다!
`background-image: url("원본이미지"}), url("실패시 대체이미지")`